### PR TITLE
[CommunityPermissions]: Added drag/drop image panel as per new design

### LINKED ISSUE #10609: [Community Mint Tokens] Update drag/drop and cropping tool according to new design
### Description

1. Extend drop area as defined in new design:
[For collectibles](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29468-610643&t=hPIthm2iHjUAucFk-0)
Screenshot 2023-05-10 at 10 00 30
[For assets](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29384-561469&t=hPIthm2iHjUAucFk-0)
Screenshot 2023-05-10 at 10 02 51

2.  When there is already an image selected:
[For collectibles](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29528%3A590023&t=hPIthm2iHjUAucFk-1)
Screenshot 2023-05-10 at 10 04 57
[For assets](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29528%3A589304&t=hPIthm2iHjUAucFk-1)
Screenshot 2023-05-10 at 10 00 58

3. Cropping tool window minimum heigh or width should be set by default fitting the minimum heigh or width of the selected image:
[For collectibles](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29468%3A611074&t=hPIthm2iHjUAucFk-1)
Screenshot 2023-05-10 at 10 19 25
[For assets](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29384%3A563130&t=hPIthm2iHjUAucFk-1)
Screenshot 2023-05-10 at 10 20 02

NOTE: Look at `CommunityNewCollectibleView` and `EditCroppedImagePanel`

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #10609

### What does the PR do
CommunityNewCollectibleView: added drag/drop image panel component for mint assets as per new designs

### Affected areas
CommunityNewCollectibleView

### Screenshot of functionality (including design for comparison)
https://github.com/status-im/status-desktop/assets/31625338/a50ab0a8-943b-4eb6-8583-044950c49a09

### CODE REVIEW

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml [hunk 1]
```diff
@@ -178,20 +233,27 @@ StatusScrollView {
 
         CustomSwitchRowComponent {
             id: transferableChecker
-
+            visible: (optionsTab.currentIndex === 0)
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:236]: We can add a constant to identify option `collectible` or `asset` to make it more readable. However, I will branch from this branch and will start the development of #10623 from here, so no need to update that. Just focus on changes related to the image and drag/drop management!

**Review by noeliaSD — COMMENTED** [review]
It looks really great!! 

I've added some comments / thoughts! Hope they are useful!

I will branch from here to start #10623 so no need to check options management. Just drag/drop and image fit!

**Review by alexjba — APPROVED** [review]
LGTM!

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml [hunk 2]
```diff
@@ -52,14 +52,15 @@ StatusScrollView {
     QtObject {
         id: d
 
+        property bool isAssetView: (optionsTab.currentIndex === 1)
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:55]: `readonly`?

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml [hunk 3]
```diff
@@ -178,20 +233,27 @@ StatusScrollView {
...
+            visible: (optionsTab.currentIndex === 0)
             label: qsTr("Remotely destructible")
             description: qsTr("Enable to allow you to destroy tokens remotely. Useful for revoking permissions from individuals")
             checked: true
         }
 
+        CustomStatusInput {
+            visible: (optionsTab.currentIndex === 1)
+            label: qsTr("Decimals")
+            secondaryLabel: "Max 10"
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:253]:
```suggestion
            secondaryLabel: qsTr("Max 10")
```

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml [hunk 4]
```diff
@@ -178,20 +233,27 @@ StatusScrollView {
...
+            placeholderText: qsTr("2")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:254]:
```suggestion
            placeholderText: "2"
```

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml (hunk 2 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:55]: I will modify it in my [pr](https://github.com/status-im/status-desktop/pull/10774).

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml (hunk 4 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:254]: Same: https://github.com/status-im/status-desktop/pull/10774

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewCollectibleView.qml:236]: Done here: https://github.com/status-im/status-desktop/pull/10774

**Review by caybro — APPROVED** [review]
LGTM then :)

**Comment by alexandraB99** [comment]
> @noeliaSD as discussed offline, for consistency it does makes sense to roll the drag and drop functionality worked on for token minting into other areas of the app where the user can upload an image. These include adding/editing the user profile image as well as the Community logo and banner.
> 
> The visual similarity of these components will lead the user to expect that the drag and drop / click to browse interaction paradigms are consistent. I will add a task to explore this in design which I can link here or create a new issue - whichever is preferable.
> 
> cc @John-44

@benjthayer please create a new issue. @noeliaSD I've created a new component for this, since it will be used in more places. when @benjthayer has finilized designs we can re-visit and apply any changes to this new component without touch the existing one. @benjthayer please have in mind when you work on the deisngs that the best thing would be to have consistency (the components looks and functions the same) among all use cases.

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            }
+        }
+    }
+
+    EditCroppedImagePanel {
+        id: editor
+        width: parent.height
+        height: width
+        Tracer {}
+        anchors.left: !root.isAssetView ? parent.left : undefined
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:64]: We should expose properties to center or anchor depending on the need.

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                editor.cropImage(drop.urls[0]);
+            }
+        }
+    }
+
+    EditCroppedImagePanel {
+        id: editor
+        width: parent.height
+        height: width
+        Tracer {}
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:63]: Leftover?

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        anchors.left: !root.isAssetView ? parent.left : undefined
+        anchors.horizontalCenter: root.isAssetView ? parent.horizontalCenter : undefined
+        visible: editor.userSelectedImage || !(root.containsDrag && root.hasImage)
+        opacity: visible ? ((root.containsDrag && root.hasImage) ? .4 : 1) : 0
+        Behavior on opacity { NumberAnimation { duration: 100 } }
+        title: qsTr("Collectible artwork")
+        acceptButtonText: qsTr("Upload collectible artwork")
+        roundedImage: root.isAssetView
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:71]: We can expose the `roundedImage` property

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        editButtonVisible: !(root.containsDrag && root.hasImage)
+
+        NoImageUploadedPanel {
+            width: parent.width
+            anchors.centerIn: parent
+            visible: !editor.userSelectedImage
+            uploadText: root.isAssetView ? qsTr("Upload") : qsTr("Drag and Drop or Upload Artwork")
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:78]: Also this upload text property

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            additionalText: qsTr("Images only")
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:79]: Maybe this too

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            showAdditionalInfo: !root.isAssetView
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:80]: Also this

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+        id: editor
+        width: parent.height
+        height: width
+        Tracer {}
+        anchors.left: !root.isAssetView ? parent.left : undefined
+        anchors.horizontalCenter: root.isAssetView ? parent.horizontalCenter : undefined
+        visible: editor.userSelectedImage || !(root.containsDrag && root.hasImage)
+        opacity: visible ? ((root.containsDrag && root.hasImage) ? .4 : 1) : 0
+        Behavior on opacity { NumberAnimation { duration: 100 } }
+        title: qsTr("Collectible artwork")
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:69]: This property should be exposed

File: ui/imports/shared/panels/DropAndEditImagePanel.qml [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+        acceptButtonText: qsTr("Upload collectible artwork")
```
Comment (noeliaSD) [inline ui/imports/shared/panels/DropAndEditImagePanel.qml:70]: And this

File: ui/imports/shared/panels/EditCroppedImagePanel.qml [hunk 13]
```diff
@@ -164,7 +167,7 @@ Item {
                 id: addButton
 
                 icon.name: "add"
-
+                visible: !dropArea.containsDrag
```
Comment (noeliaSD) [inline ui/imports/shared/panels/EditCroppedImagePanel.qml:170]: `dropArea` component has been removed. There is also reference of it in line `157` and `161`. States are not needed now!

**Review by noeliaSD — COMMENTED** [review]
Added some comments about new component `DropAndEditImagePanel`.

**Pull request merged**